### ISSUE TITLE
[Critical some questions] Tensorflow-lite, Neural Networks API

### ISSUE THREAD

**Issue by nanamare** [tensorflow/tensorflow#16979]
Hi.

**Some question.**

 **1.**

> Current i use  tensorflow library `  compile 'org.tensorflow:tensorflow-lite:+'`
> and i want to use Neural Networks Api, to reduce inference time.
> 
> but wrapper function is not existing in `Interpreter.java` class 
> `  private static native void useNNAPI(long var0, boolean var2);`
> so i can't use it..
> 
> If I create a wrapper function, can I use Neural Networks API? (my device level >  8.1)

What is https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/java/src/testhelper/java/org/tensorflow/lite/TestHelper.java
?

**2.**

> i'm converting pb to tflite. mobilenet_v1_224.pb (17 mb) mobilenet_v1_224_uint8.tflite (4 mb) 
> inference speed about 290 ms -> 73ms decreased

> Command line(mobileNet)

<pre><code>
bazel run -c opt --copt=-msse4.1 --copt=-msse4.2 --config=opt \
  //tensorflow/contrib/lite/toco:toco -- \
  --input_file=/home/danshin/tensorflow_lite/lite_model/frozen_graph.pb \
  --output_file=/home/danshin/tensorflow_lite/lite_model/frozen_uint_graph.tflite \
  --input_format=TENSORFLOW_GRAPHDEF \
  --output_format=TFLITE \
  --inference_type=QUANTIZED_UINT8 \
  --input_shape=1,224,224,3 \
  --input_array=input \
  --output_array=MobilenetV1/Predictions/Reshape_1 \
  --default_ranges_min=0 \
  --default_ranges_max=6 \
  --mean_value=127.5 \
  --std_value=127.5
</pre></code>

> In a similar way,
> But frozen_cpm.pb(120 mb) convert to frozen_cpm_uint8.tflite(30 mb)
> inference speed about 11000 ms -> 11000ms. 
>  **If the model size is reduced, does not the inference time generally decrease?**

**Comment by nanamare** [comment]
OS Platform and Distribution :  Ubuntu 16.04 LTS
TensorFlow installed from : official
TensorFlow version : 1.5 version
Bazel version : 0.9.0 
CUDA/cuDNN version : V 8.0.61 (8.0)
GPU model and memory : GeForce GTX 1060 / 3G
Exact command to reproduce : 
```
bazel run -c opt --copt=-msse4.1 --copt=-msse4.2 --config=opt \
  //tensorflow/contrib/lite/toco:toco -- \
  --input_file=/home/danshin/tensorflow_lite/lite_model/frozen_cpm.pb \
  --output_file=/home/danshin/tensorflow_lite/lite_model/frozen_uint8_cpm.tflite \
  --input_format=TENSORFLOW_GRAPHDEF \
  --output_format=TFLITE \
  --input_shape=1,368,368,3 \
  --inference_type=QUANTIZED_UINT8 \
  --input_array=inference_images/Placeholder \
  --output_array=stage6/confidence_maps/BiasAdd \
  --logtostderr
```

**Comment by nanamare** [comment]
@cy89 , @aselle

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by andrehentz** [comment]
I think there's already an official way to turn on the NNAPI in the JAVA Api of TF Lite. It will not make much of a difference unless your device provides NNAPI acceleration.
